crates/wasi-nn/src/backend/winml.rs

```diff
@@ -175,23 +158,19 @@ impl BackendExecutionContext for WinMLExecutionContext {
         if let Some(result) = &self.result {
             let output_features = self.session.Model()?.OutputFeatures()?;
             let index = self.find(id, &output_features)?;
-            let output = output_features.GetAt(index)?;
-            // TODO: this only handles FP32!
-            let tensor = result
-                .Outputs()?
-                .Lookup(&output.Name()?)?
-                .cast::<TensorFloat>()?;
-            let dimensions = dimensions_as_u32(&tensor.Shape()?)?;
-            let view = tensor.GetAsVectorView()?;
-            let mut data = Vec::with_capacity(view.Size()? as usize * size_of::<f32>());
-            for f in view.into_iter() {
-                data.extend(f.to_le_bytes());
-            }
-            Ok(Tensor {
-                ty: TensorType::Fp32,
-                dimensions,
-                data,
-            })
+            let output_feature = output_features.GetAt(index)?;
+            let tensor_kind = match output_feature.Kind()? {
+                windows::AI::MachineLearning::LearningModelFeatureKind::Tensor => output_feature
+                    .cast::<TensorFeatureDescriptor>()?
```

Comment:
Are we sure `cast` is right here? Poking around in `windows-rs`, I see that it has a `CanInto` implementation for `TensorFeatureDescriptor` [here](https://microsoft.github.io/windows-docs-rs/doc/windows/AI/MachineLearning/struct.ILearningModelFeatureDescriptor.html#impl-CanInto%3CILearningModelFeatureDescriptor%3E-for-TensorFeatureDescriptor) &mdash; doesn't that mean we should be using `into()` instead?

Reply:
My understanding is `cast` here is for COM interfaces, similar as [QueryInterface](https://learn.microsoft.com/en-us/cpp/atl/queryinterface), while `into` is for converting derived class into base class?